### PR TITLE
fixes #18 - implement reconnecting and pinging websocket

### LINKED ISSUE #18: Implement reconnect mechanism for websockets
Right now, the websockets have no reconnect mechanism. AWS' websockets hang up (always!!) after 10 minutes on no messages, and after 2 hours regardless.

This means the live page stops working (timers just keep running and the page needs to be refreshed) after 2 hours.

Implement a reconnect when the server hangs up.

### PR DESCRIPTION
**Pull request by therungg**
Creates a new hook that wraps `useWebsocket` that does two things:

- Reconnects automatically with exponential backoff after disconnect
- Pings the server every 9 minutes to prevent the server automatically closing

Both to deal with AWS hard limits: https://docs.aws.amazon.com/apigateway/latest/developerguide/limits.html

### CODE REVIEW

**Review by francis-robert — APPROVED** [review]
Really great work! No comments on this just a few questions:

- Is the username the only argument that would ever be wanted? Could there ever be a need for others in the future?
- Is the last message the only return value that will be needed?

**Comment by therungg** [comment]
> Really great work! No comments on this just a few questions:
> 
>     * Is the username the only argument that would ever be wanted? Could there ever be a need for others in the future?
> 
>     * Is the last message the only return value that will be needed?

Thanks for looking at it, appreciate it!

- For now, the username is the only filter on the websocket api. In the future, there *might* be more, but I prefer to keep it simple and adjust later in one go if I need more filters.
- I don't think there will be another value needed. The websocket will probably always be passive, so `sendMessage` will not be needed. I also don't see a reason for the implementing code to close the client, or access the ready state, since the latter can be infered from the `lastMessage`.

So in short: extensions *might* be needed in the future, but I'd rather deal with it then and keep it simple for now.

Going to merge :D
